### ISSUE TITLE
Numerical differences

### ISSUE THREAD

**Issue by txy00001** [ultralytics/ultralytics#14264]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Why do I load #model=YOLO("yolov8m-seg.pt") directly, and the result is YOLOv8m-seg summary: 331 layers, 27243701 parameters, 27243685 gradients, 110.4 GFLOPs, and I model = YOLO("ultralytics/cfg/models/v8/ yolov8-seg.yaml").load("yolov8m-seg.pt"), the result is YOLOv8-seg summary: 261 layers, 3264981 parameters, 3264965 gradients, 12.1 GFLOPs, why is that?

**Comment by glenn-jocher** [comment]
@txy00001 hello,

Thank you for reaching out with your question! The discrepancy you're observing between the two methods of loading the model is due to the different configurations and layers being initialized.

When you load the model directly using `model = YOLO("yolov8m-seg.pt")`, it loads the pre-trained weights and the architecture as defined in the `.pt` file. This includes all the layers and parameters that were used during the training of this specific model.

On the other hand, when you use `model = YOLO("ultralytics/cfg/models/v8/yolov8-seg.yaml").load("yolov8m-seg.pt")`, you are first initializing the model architecture from the YAML configuration file and then loading the weights from the `.pt` file. The YAML file might define a different architecture or omit certain layers that are present in the pre-trained model, leading to the differences in the number of layers, parameters, gradients, and GFLOPs.

To ensure consistency, it's recommended to load the model directly from the `.pt` file if you want to use the exact architecture and parameters as the pre-trained model. Here’s a concise example for clarity:

```python
from ultralytics import YOLO

# Recommended way to load the pre-trained model
model = YOLO("yolov8m-seg.pt")
```

If you believe there might be an issue with the YAML configuration or if you need to customize the model architecture, please ensure that the YAML file accurately reflects the architecture used in the pre-trained model.

For further assistance, providing a minimum reproducible example can be very helpful. You can find more details on how to create one here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Feel free to update to the latest version of the packages and verify if the issue persists. If you have any more questions or need further clarification, don't hesitate to ask! 😊
